sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SymbolicExecution/AbstractExplodedGraph.cs

```diff
@@ -358,14 +360,24 @@ protected void EnqueueNewNode(ProgramPoint programPoint, ProgramState programSta
                 this.programPoints[pos] = pos;
             }
 
+            ExplodedGraphNode newNode = null;
             if (programState.GetVisitedCount(pos) >= MaxProgramPointExecutionCount)
             {
                 OnProgramPointVisitCountExceedLimit(pos, programState);
-                return;
+
+                // reached the max number of visit by program point, so we take the foreach loop false branch with current program state

```

Comment:
given that this gets executed for everything, not only `ForEach` blocks, I would change the comment to emphasize that if *might* be a for each...

and also please add a comment that if it is _not_ a foreach block AND it has reached the MaxNumberOfSteps, the `newNode` will be null, thus it will end the exploration